core/src/main/java/brooklyn/entity/rebind/RebindIteration.java

```diff
@@ -0,0 +1,1086 @@
+        this.persistenceStoreAccess = persistenceStoreAccess;
+        
+        managementContext = rebindManager.getManagementContext();
+        rebindContext = new RebindContextImpl(exceptionHandler, classLoader);
+        reflections = new Reflections(classLoader);
+        lookupContext = new RebindContextLookupContext(managementContext, rebindContext, exceptionHandler);
+        rebindContext.setLookupContext(lookupContext);
+        instantiator = new BrooklynObjectInstantiator(classLoader, rebindContext, reflections);
+        
+        if (mode==ManagementNodeState.HOT_STANDBY || mode==ManagementNodeState.HOT_BACKUP) {
+            rebindContext.setAllReadOnly();
+        } else {
+            Preconditions.checkState(mode==ManagementNodeState.MASTER, "Must be either master or read only to rebind (mode "+mode+")");
+        }
+    }
+
+    public List<Application> getApplications() {
+        return applications;
+    }
+    
+    RebindContextImpl getRebindContext() {
+        return rebindContext;
+    }
+    
+    public void runFullRebind() {
+        runWithLock(new Callable<Void>() {
+           public Void call() throws Exception {
+               
+               loadManifestFiles();
+               rebuildCatalog();
+               instantiateLocationsAndEntities();
+               instantiateMementos();
+               instantiateAdjuncts(instantiator); 
+               reconstructEverything();
+               associateAdjunctsWithEntities();
+               manageTheObjects();
+               finishingUp();
+               
+               return null;
+           }
+        });
+    }
+    
+    protected void runWithLock(Callable<?> target) {
+        if (iterationStarted.getAndSet(true)) {
+            throw new IllegalStateException("Iteration "+this+" has already run; create a new instance for another rebind pass.");
+        }
+        try {
+            rebindActive.acquire();
+        } catch (InterruptedException e) { Exceptions.propagate(e); }
+        try {
+            RebindTracker.setRebinding();
+            if (ManagementNodeState.isHotProxy(mode))
+                readOnlyRebindCount.incrementAndGet();
+
+            timer = Stopwatch.createStarted();
+            exceptionHandler.onStart(rebindContext);
+
+            target.call();
+            
+            exceptionHandler.onDone();
+            
+            rebindMetrics.noteSuccess(Duration.of(timer));
+            noteErrors(exceptionHandler, null);
+            
+        } catch (Exception e) {
+            rebindMetrics.noteFailure(Duration.of(timer));
+            
+            Exceptions.propagateIfFatal(e);
+            noteErrors(exceptionHandler, e);
+            throw exceptionHandler.onFailed(e);
+            
+        } finally {
+            rebindActive.release();
+            RebindTracker.reset();
+        }
+    }
+    
+    protected void checkEnteringPhase(int targetPhase) {
+        phase++;
```

Comment:
Maybe should use an enum, with ordinal for readability of these calls? No strong feelings.